src/gui_common/CellHexPhotoBuilder.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using Godot;
+
+public class CellHexPhotoBuilder : Spatial, IPhotographable
+{
+    private float radius;
+    private bool radiusDirty;
+    private MicrobeSpecies? species;
+
+    public string SceneToPhotographPath => "res://src/gui_common/CellHexPhotoBuilder.tscn";
+
+    public float Radius
+    {
+        get
+        {
+            if (radiusDirty)
+                CalculateRadius();
+
+            return radius;
+        }
+    }
+
+    public MicrobeSpecies? Species
+    {
+        get => species;
+        set
+        {
+            species = value;
+            radiusDirty = true;
+        }
+    }
+
+    public void ApplySceneParameters(Spatial instancedScene)
+    {
+        var builder = (CellHexPhotoBuilder)instancedScene;
+        builder.Species = Species;
+        builder.BuildHexStruct();
+    }
+
+    public float CalculatePhotographDistance(Spatial instancedScene)
+    {
+        return PhotoStudio.CameraDistanceFromRadiusOfObject(((CellHexPhotoBuilder)instancedScene).Radius *
+            Constants.PHOTO_STUDIO_CELL_RADIUS_MULTIPLIER);
+    }
+
+    private void BuildHexStruct()
+    {
+        var hexScene = GD.Load<PackedScene>("res://src/microbe_stage/editor/EditorHex.tscn");
+        var validMaterial = GD.Load<Material>("res://src/microbe_stage/editor/ValidHex.material");
+        var modelScene = GD.Load<PackedScene>("res://src/general/SceneDisplayer.tscn");
+
+        foreach (Node node in GetChildren())
+            node.DetachAndQueueFree();
```

Comment:
I'm pretty sure there's an extension method to free all child nodes in NodeHelpers or something.